File: app/src/main/java/me/dev4vin/samplepromiseapp/auth/Role.java
package me.dev4vin.samplepromiseapp.auth;

public class Role {
  public static final String CAN_ADD_TODO = "can_add_todo", CAN_UPDATE_TODO = "can_update_todo";

  private String name;
  private boolean allowed;

  public String name() {
    return name;
  }

  public Role name(String name) {
    this.name = name;
    return this;
  }

  public boolean allowed() {
    return allowed;
  }

  public Role allowed(boolean allowed) {
    this.allowed = allowed;
    return this;
  }
}

File: app/src/main/java/promise/app/auth/Role.kt
package promise.app.auth

class Role {

  private var name: String? = null
  private var allowed: Boolean = false

  fun name(): String? {
    return name
  }

  fun name(name: String): Role {
    this.name = name
    return this
  }

  fun allowed(): Boolean {
    return allowed
  }

  fun allowed(allowed: Boolean): Role {
    this.allowed = allowed
    return this
  }

  companion object {
    val CAN_ADD_TODO = "can_add_todo"
    val CAN_UPDATE_TODO = "can_update_todo"
  }
}
